surface_potential_analysis/kernel/build.py

```diff
@@ -95,7 +92,38 @@ def build_isotropic_kernel_from_function(
     displacements = get_displacements_x(basis)
     correlation = fn(displacements["data"].reshape(displacements["basis"].shape)[0])
 
-    return {"basis": displacements["basis"][0], "data": correlation.ravel()}
+    return {"basis": displacements["basis"][0][0], "data": correlation.ravel()}
+
+
+def build_isotropic_kernels_from_function(
+    basis: StackedBasisWithVolumeLike[Any, Any, Any],
+    fn: Callable[
+        [np.ndarray[Any, np.dtype[np.float64]]],
+        np.ndarray[Any, np.dtype[np.complex128]],
+    ],
+) -> tuple[
+    IsotropicNoiseKernel[FundamentalPositionBasis[Any, Any]],
+    ...,
+]:
+    """
+    Get an Isotropic Kernel with a correllation beta(x-x').
+
+    Parameters
+    ----------
+    basis : StackedBasisWithVolumeLike[Any, Any, Any]
+    fn : Callable[
+        [np.ndarray[Any, np.dtype[np.float64]]],
+        np.ndarray[Any, np.dtype[np.complex128]],
+    ]
+        beta(x-x'), the correllation as a function of displacement
+
+    Returns
+    -------
+    IsotropicNoiseKernel[
+        TupleBasisWithLengthLike[*tuple[FundamentalPositionBasis[Any, Any], ...]],
+    ]
+    """
+    return tuple(build_isotropic_kernel_from_function(basis_i, fn) for basis_i in basis)
```

Comment:
add a build_isotropic_kernel() which works on a flat axis to sort these type issues, also add a build_isotropic_kernel_stacked_from_function which gets the full stacked kernel like build_isotropic_kernel_from_function currently does